subsys/net/lib/openthread/platform/uart.c

```diff
@@ -197,23 +192,65 @@ otError otPlatUartEnable(void)
 
 otError otPlatUartDisable(void)
 {
-	/* TODO: uninit UART */
-	LOG_WRN("%s not implemented.", __func__);
-	return OT_ERROR_NOT_IMPLEMENTED;
+#ifdef CONFIG_OPENTHREAD_NCP_SPINEL_ON_UART_ACM
+	int ret = usb_disable();
+	if (ret) {
+		LOG_WRN("Failed to disable USB, ret code %d", ret);
+	}
+#endif
+
+	uart_irq_tx_disable(ot_uart.dev);
+	uart_irq_rx_disable(ot_uart.dev);
+	return OT_ERROR_NONE;
 };
 
 
 otError otPlatUartSend(const u8_t *aBuf, u16_t aBufLength)
 {
-	size_t cnt = ring_buf_put(ot_uart.tx_ringbuf, aBuf, aBufLength);
+	if (NULL == aBuf) {
+		return OT_ERROR_FAILED;
+	}
+
+	sWriteBuffer = aBuf;
```

Comment:
The transmission is asynchronous, does OT guarantee that the buffer won't be violated in any way before transmission finishes (I assume so given we have `otPlatUartSendDone`)? If so, I'd emphasize this in the commit message.

Reply:
Yes, buffer cannot be reused until send done notification method is called. I mentioned it in commit message.